designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,213 @@
+
+## Proposal
+
+To reduce the user adaptation, the command
+leverage the existing kustomize default directory and create the similar layout of the helm chart from it. The generated `values.yaml` is as follows:
+
+```yaml
+# global configurations
+nameOverride: ""
+
+fullnameOverride: ""
+
+crd:
+  create: true
+
+## More info: https://kubernetes.io/docs/admin/authorization/rbac/
+rbac:
+  create: true
+  serviceAccountName: ""
+
+
+## Configure the manager
+manager:
+  ## The Security Context of the manager Pod.
+  ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+  podSecurityContext:
+  runAsNonRoot: true
+
+  ## Configure the kubeRbacProxy container.
+  kubeRbacProxy:
+    image:
+     registry: gcr.io
+     repository: "kubebuilder/kube-rbac-proxy"
+     tag: v0.13.0
+     pullPolicy: IfNotPresent
+
+  ## Configure the resources.
+  ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+  resources:
+    limits:
+      cpu: 500m
+      memory: 128Mi
+    requests:
+      cpu: 5m
+      memory: 64Mi
+
+  ## Configure the controller container.
+  controller:
+    image:
+      registry: docker.io
+      repository: ""
+      tag: latest
+      pullPolicy: IfNotPresent
+
+     ## The environment variables of the controller container.
+     ## More info: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
+  	env: []
+
+    ## Configure extra options for liveness probe.
+    ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
+    livenessProbe:
+      enabled: true
+      httpGet:
+        path: /healthz
+        port: 8081
+      initialDelaySeconds: 15
+      periodSeconds: 20
+
+    readinessProbe:
+      enabled: true
+      httpGet:
+        path: /readyz
+        port: 8081
+      initialDelaySeconds: 5
+      periodSeconds: 10
+
+    ## Configure the resources.
+    ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+    resources:
+      limits:
+        cpu: 500m
+        memory: 128Mi
+      requests:
+        cpu: 10m
+        memory: 64Mi
+
+     ## Configure the scheduling rules.
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+     affinity: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+     nodeSelector: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration
+     tolerations: {}
+
+     ## Configure the rbac roles.
+     serviceAccountName: ""
+
+## Configure the webhook
+webhook:
+  enabled: true
+  ## More info: https://kubernetes.io/docs/concepts/services-networking/service/
+  service:
+    type: ClusterIP
+    ports:
+    - port: 443
+      protocol: TCP
+      targetPort: 9443
```

Comment:
This one would be `config/default/manager_webhook_patch.yam` : https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/config/default/manager_webhook_patch.yaml#L6-L23